browser/src/impl.ts

```diff
@@ -31,73 +40,209 @@ export type ChainInfo = {
+export async function createQuarry(
   network: Network,
   options: ClientOptions
-): QuarryClient {
+): Promise<QuarryClient> {
+  // The latest block header.
+  let head: BlockHeader | null = null;
+
+  // insecure keystore in memory for development. Do not store real private keys in there!
   const keystore: Map<string, Key> = new Map();
-  return {
-    onPeerConnected: function (cb) {
-      network.connectionManager.addEventListener("peer:connect", (conn) =>
-        cb(conn.detail.remoteAddr)
-      );
+  const msgTopic = "/fil/msgs/" + options.networkName;
+  const blkTopic = "/fil/blocks/" + options.networkName;
+
+  // gossip params are setup after Lotus params in order for the node to be treated
+  // as similarly as other peers as possible. Need more research into fine tuning these.
+  const pubsub = new GossipSub(network, {
+    floodPublish: true,
+    allowedTopics: [msgTopic, blkTopic],
+    scoreParams: {
+      topics: {
+        [blkTopic]: {
+          ...defaultTopicScoreParams,
+          // expected 10 blocks/min
+          topicWeight: 0.1, // max cap is 50, max mesh penalty is -10, single invalid message is -100
+          // 1 tick per second, maxes at 1 after 1 hour
+          timeInMeshWeight: 0.00027, // ~1/3600
+          timeInMeshQuantum: 1,
+          timeInMeshCap: 1,
+          // deliveries decay after 1 hour, cap at 100 blocks
+          firstMessageDeliveriesWeight: 5, // max value is 500
+          firstMessageDeliveriesDecay: 0.998722,
+          firstMessageDeliveriesCap: 100, // 100 blocks in an hour
+          // invalid messages decay after 1 hour
+          invalidMessageDeliveriesWeight: -1000,
+          invalidMessageDeliveriesDecay: 0.998722,
+        },
+        [msgTopic]: {
+          ...defaultTopicScoreParams,
+          // expected > 1 tx/second
+          topicWeight: 0.1, // max cap is 5, single invalid message is -100
+          // 1 tick per second, maxes at 1 hour
+          timeInMeshWeight: 0.0002778, // ~1/3600
+          timeInMeshQuantum: 1,
+          timeInMeshCap: 1,
+          // deliveries decay after 10min, cap at 100 tx
+          firstMessageDeliveriesWeight: 0.5, // max value is 50
+          firstMessageDeliveriesDecay: 0.992354,
+          firstMessageDeliveriesCap: 100, // 100 messages in 10 minutes
+          // invalid messages decay after 1 hour
+          invalidMessageDeliveriesWeight: -1000,
+          invalidMessageDeliveriesDecay: 0.998722,
+        },
+      },
+      topicScoreCap: 10.0,
+      // Can prevent pruning bootstrap nodes when we know them.
+      appSpecificScore: () => 0.0,
+      appSpecificWeight: 1,
+      IPColocationFactorWeight: -100,
+      IPColocationFactorThreshold: 5,
+      IPColocationFactorWhitelist: new Set(),
+      behaviourPenaltyWeight: -10.0,
+      behaviourPenaltyThreshold: 6,
+      behaviourPenaltyDecay: 0.998722,
+      decayInterval: 1000.0,
+      decayToZero: 0.1,
+      retainScore: 3600 * 6,
```

Comment:
@alexander-camuto here are all the params. Pretty much mirrored everything in [lotus](https://github.com/filecoin-project/lotus/blob/c602412eba07c3e5650b6ebd59d93f640411ee1b/node/modules/lp2p/pubsub.go#L76). I don't quite understand every param yet.

Reply:
I don't think these will be our salvation in terms of preventing the browser peer from getting booted, as these just pertain to scoring the _other_ peers we interact with. We'll maybe have to play around with some of the other `GossipsubOpts` ?